crates/rover-client/src/error.rs

```diff
@@ -0,0 +1,30 @@
+use thiserror::Error;
+
+/// An enum represent all type of possible failures.
+#[derive(Error, Debug)]
+pub enum RoverClientError {
+    /// The provided GraphQL was invalid.
+    #[error("encountered a GraphQL error, registry responded with: {msg}")]
+    GraphQLError {
```

Comment:
```suggestion
    GraphQL {
```

generally safe to leave `Error` out of these names since that's implied

Reply:
if you use an invalid API token, you get this message, which is not super great `encountered a GraphQL error, registry responded with: 406: Not Acceptable`

should figure out how we can make it inform you that your API key is the problem